sql/inspect: implement INSPECT processor stub

spilchen opened the pull request:
This commit completes the implementation of the INSPECT processor, which executes the per-node portion of an INSPECT job. The processor follows a producer-worker model: the producer enqueues spans, which are consumed by multiple concurrent workers. By default, the number of workers matches the output of runtime.GOMAXPROCS(0), but it can be overridden via the sql.inspect.processor_concurrency setting.

The processor uses inspectRunner to perform various checks. This commit introduces a stub for an index consistency check and a testing-only check to validate processor logic.

Informs: #148299

Epic: CRDB-30356

Release note: none

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/149894)
<!-- Reviewable:end -->

spilchen committed:
sql/inspect: implement INSPECT processor stub

This commit completes the implementation of the INSPECT processor, which
executes the per-node portion of an INSPECT job. The processor follows a
producer-worker model: the producer enqueues spans, which are consumed by
multiple concurrent workers. By default, the number of workers matches the
output of runtime.GOMAXPROCS(0), but it can be overridden via the
sql.inspect.processor_concurrency setting.

The processor uses inspectRunner to perform various checks. This commit
introduces a stub for an index consistency check and a testing-only check to
validate processor logic.

Informs: #148299

Epic: CRDB-30356

Release note: none

rafiss reviewed (approved):
nice, these interfaces made it very easy to review. lgtm!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/149894#-:-OVO8-hX9xOnGiQxYj0Z:b-9dzjld)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @spilchen)



<!-- Sent from Reviewable.io -->
